File: src/1232. Check If It Is a Straight Line.js
/**
 * @param {number[][]} coordinates
 * @return {boolean}
 */

const checkStraightLine = (coordinates) => {
  const n = coordinates.length
  if (n <= 1) return false
  const [x1, y1] = coordinates[0]
  const [x2, y2] = coordinates[1]
  for (let i = 2; i < n; i++) {
    let [x3, y3] = coordinates[i]
    if ((y1 - y2) * (x2 - x3) !== (y2 - y3) * (x1 - x2)) return false
  }
  return true
}

File: src/1232. Check If It Is a Straight Line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        const int dx1 = coordinates[1][0] - coordinates[0][0], dy1 = coordinates[1][1] - coordinates[0][1];
        for (int i = 2; i < coordinates.size(); ++i) {
            const int dx2 = coordinates[i][0] - coordinates[0][0], dy2 = coordinates[i][1] - coordinates[0][1];
            if (dx1 * dy2 != dx2 * dy1) {
                return false;
            }
        }
        return true;
    }
};